### DISCUSSION TITLE
Get a webhook when a GitHub Actions run is queued

### DISCUSSION THREAD

**Discussion by mrcasals** [community/community#24936] [Actions]
<p>Is this possible? I couldn’t find anything on the docs, but I think it would be useful! I’m thinking about building some in-house service that cancels redundant builds, but for that I’d need to keep the run IDs somewhere… I could ping the API every X minutes to get the latest info, but this doesn’t escale much…</p>
<p>Thanks!</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/mrcasals">@mrcasals</a> ,</p>
<p>There is not such webhook in github actions.</p>
<p>All webhook events are mentioned in <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows" rel="nofollow noopener">doc </a>and job info can be got from <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions" rel="nofollow noopener">context</a>(eg: ${{ job.status }}). Typpically we will use if expression to check and execute the next step accordingly.</p>
<p>Thanks.</p>

**Comment by mrcasals** [comment]
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a>!</p>
<p>The docs you link to are for webhooks that trigger workflows. I’d like th other way around: to be notified every time a workflow run is enqueued. Is that possible?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/mrcasals">@mrcasals</a> ,</p>
<p>The only related webhooks i can find are in “repo settings/hooks”, and guide <a href="https://developer.github.com/webhooks/" rel="nofollow noopener">here.</a></p>
<p><img src="https://user-images.githubusercontent.com/52772761/181087821-5eed3662-b17e-4bc8-a4a4-d3707f94cd5f.png" alt="webhook3.png" data-base62-sha1="nYfSsbUI0xZONvc8LiyJolXBRtO" width="617" height="120"></p>
<p>Is this you’re looking for? </p>
<p>Thanks.</p>

**Comment by mrcasals** [comment]
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> I’m super late to this, I missed the notification, but yes, this seems to be exactly what I need! Thanks!</p>
